pkg/pcr/get_measurements.go

```diff
@@ -53,3 +52,133 @@ func GetMeasurements(
 	}
 	return
 }
+
+func getMeasurements(
+	pcrID ID,
+	firmware Firmware,
+	config MeasurementConfig,
+) (
+	Measurements,
+	Flow,
+	map[string]interface{},
+	error,
+) {
+	resultConfig := config
+	debugInfo := map[string]interface{}{
+		"config_orig":   config,
+		"config_result": &resultConfig,
+	}
+
+	// Detect attestation flow
+	detectedFlow, detectFlowErr := DetectAttestationFlow(firmware, config.Registers, config.TPMDevice)
+	if detectedFlow != FlowAuto {
+		debugInfo["detectedAttestationFlow"] = detectedFlow.String()
```

Comment:
nit
As we are doing a little bit of refactor, I would also rethink debugInfo. Make all potential keys as public constants should be a good start